### DISCUSSION TITLE
HLS and cross-compiler support - what is the current state?

### DISCUSSION THREAD

**Discussion by alaendle** [haskell/haskell-language-server#4187] [Q&A] [answered]
Hi,

please apologize if I have missed something, but I'm not sure if there is a way to get HLS running in conjunction with a cross-compiled GHC (e.g. to use the javascript backend). 

I failed and I tracked this down to - regardless of HIE configuration, etc. - HLS always tries to use GHC interactively - which requires a stage2 compiler.
https://github.com/haskell/haskell-language-server/blob/f8379bb42c6e1e647e25ee2dec454e47ff544d8a/exe/Wrapper.hs#L169

As far as I interpret https://gitlab.haskell.org/ghc/ghc/-/issues/19174 and related issues to compile a cross compiler to stage2 and therefor it is currently not possible to have HLS working with the javascript backend?

Or is it somehow possible to run HLS in conjunction with the GHC javascript compiler?

Thanks in advance for any hint!

Best regards,

Andreas

**Comment by michaelpj** [comment, marked as answer]
I think the answer is probably "nobody has really tried". I suspect there are many places where we rely on things that are tricky in cross-compiler territory. Certainly, we rely on calling `cabal repl` to get compilation flags, I don't even know if that works.

**Reply by alaendle** [reply]
I should have phrased my question more clearly. Actually I just wanted to know if anyone had had success getting HLS to work with a cross compiler. I tried it and failed. Maybe it's better to just wait until GHC produces better cross compilers (keep fingers crossed for ghc-9.12).
